cli/src/data.ts

```diff
@@ -59,25 +66,79 @@ function loadTinderDogData(split: number): Dataset<DataFormat.Raw["image"]> {
     });
 }
 
+async function readLabels(labelsPath:string): Promise<string[]>{
+  const raw = await fs.readFile(labelsPath, "utf8");
+  return raw.split(/\r?\n/).map(s => s.trim()).filter(s => s.length > 0);
+}
+
+function loadExtCifar10(userIdx: number){
+  const CIFAR10_LABELS = ["airplane", "automobile", "bird", "cat", "deer", "dog", "frog", "horse", "ship", "truck"];
+  const folder = path.join("..", "datasets", "extended_cifar10");
+  const clientFolder = path.join(folder, `client_${userIdx}`);
+
+  return new Dataset<[Image, string]>(async function*(){
+    const entries = await fs.readdir(clientFolder, {withFileTypes: true});
+
+    const items = entries
+        .map((e) => {
+          const m = e.name.match(
+            /^image_(\d+)_label_(\d+)\.png$/i
+          );
+          if (!m) return null;
+
+          const imgIdx = Number(m[1]);
+          const labelIdx = Number(m[2]);
```

Comment:
better to use `Number.parseInt` as to avoid the weird coercions rules of js

```suggestion
          const imgIdx = Number.parseInt(m[1], 10);
          const labelIdx = Number.parseInt(m[2], 10);
```